apidef/oas/upstream.go

```diff
@@ -191,3 +195,41 @@ func (t *Test) ExtractTo(uptimeTests *apidef.UptimeTests) {
 		t.ServiceDiscovery.ExtractTo(&uptimeTests.Config.ServiceDiscovery)
 	}
 }
+
+type MutualTLS struct {
+	Enabled       bool           `bson:"enabled" json:"enabled"`
+	DomainToCerts []DomainToCert `bson:"domainToCertificateMapping" json:"domainToCertificateMapping"`
+}
+
+type DomainToCert struct {
+	Domain string `bson:"domain" json:"domain"`
+	Cert   string `bson:"certificate" json:"certificate"`
+}
+
+func (m *MutualTLS) Fill(api apidef.APIDefinition) {
+	m.Enabled = !api.UpstreamCertificatesDisabled
+
+	if len(api.UpstreamCertificates) == 0 {
```

Comment:
This check may be redundant, we may always want to allocate `DomainToCerts` even with a 0-length, so it will pass OAS validation (type: array).